Userland/Utilities/tail.cpp

```diff
@@ -102,12 +103,32 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     TRY(tail_from_pos(*f, pos));
 
     if (follow) {
+        file_size = TRY(f->size());
         TRY(f->seek(0, SeekMode::FromEndPosition));
 
         Core::EventLoop event_loop;
         auto watcher = TRY(Core::FileWatcher::create());
         watcher->on_change = [&](Core::FileWatcherEvent const& event) {
             if (event.type == Core::FileWatcherEvent::Type::ContentModified) {
+                auto maybe_current_size = f->size();
+                if (maybe_current_size.is_error()) {
+                    auto error = maybe_current_size.release_error();
+                    warnln(error.string_literal());
+                    event_loop.quit(error.code());
```

Comment:
This error handling code is repeated 4 times, maybe put it in a lambda?

Reply:
lambdas are not my strong suite, but I'll take a look.